### DISCUSSION TITLE
Is it safe to add a Bearer token to the QueryClient cache?

### DISCUSSION THREAD

**Discussion by Victor-Villacis** [TanStack/query#4084] [Q&A]
I wanted to provide headers that pass the functions 
getIdToken()     
getJwtToken()

Which authorizes and passes in my Bearer token in order validate my api that I have set up in AWS. That way its available throughout the application. What is the best practice in which this is done?

**Comment by TkDodo** [comment]
tokens and authentication is usually something that you handle on the layer that actually makes the request (axios, fetch, whatever). There is no need to involve react-query.

**Reply by Victor-Villacis** [reply]
Hmm correct, I wanted to pass the auth globally though so I don't constantly need to add it every time I make a request. Was just wondering if there was any concerns on doing so.
